precommit/src/main/shell/core.d/change-analysis.sh

```diff
@@ -68,7 +68,9 @@ function find_changed_files
       # Additionally, remove any a/ b/ patterns at the front of the patch filenames.
       # shellcheck disable=SC2016
       while read -r line; do
-        CHANGED_FILES=("${CHANGED_FILES[@]}" "${line}")
+        if [[ -n "${line}" ]]; then
+          CHANGED_FILES=("${CHANGED_FILES[@]}" "${line}")
+        fi
       done < <(
         "${AWK}" 'function p(s){sub("^[ab]/","",s); if(s!~"^/dev/null"){print s}}
```

Comment:
If "---" or "+++" is passed, the awk function prints an empty string. I think this issue should be fixed in the function.

```suggestion
        "${AWK}" 'function p(s){sub("^[ab]/","",s); if(s!~"^/dev/null"&&s!~"^[[:blank:]]*$"){print s}}
```

Reply:
Awesome! Thanks.  Found another place where this awk construction is used so updated it too.